plugins/module_utils/redfish_utils.py

```diff
@@ -819,7 +819,11 @@ def get_disk_inventory(self, systems_uri):
                             for property in properties:
                                 if property in data:
                                     if data[property] is not None:
-                                        drive_result[property] = data[property]
+                                        if property == "Links" and "Volumes" in data["Links"].keys():
```

Comment:
This looks like it will still add "Links" if "Volumes" isn't inside of "Links". So, if you have a response like this:

```
{
    "Id": "Drive0",
    "Name": "Drive Slot 0",
    ...
    "Links": {
        "Chassis": { "@odata.id": "/redfish/v1/Chassis/1" }
    }
}
```

it will produce an output like this:

```
Id: Drive0
Name: Drive Slot 0
Links:
  Chassis:
    - @odata.id: /redfish/v1/Chassis/1
```

Could we break out this if statement to prevent unwanted links from appearing like this?

Reply:
Updated as per suggestion.

Reply:
Resolving this conversation. Thanks!